OutOfSchool/OutOfSchool.WebApi.Tests/Controllers/ProviderControllerTests.cs

```diff
@@ -1,519 +1,290 @@
+        private ProviderDto providerDto;
+
+        [SetUp]
+        public void Setup()
+        {
+
+            var localizer = new Mock<IStringLocalizer<SharedResource>>();
+            var user = new ClaimsPrincipal
+                (new ClaimsIdentity(
+                    new Claim[]
+                    {
+                        new Claim(IdentityResourceClaimsTypes.Sub, Guid.NewGuid().ToString()),
+                        new Claim(IdentityResourceClaimsTypes.Role, Role.Provider.ToString()),
+                    },
+                    IdentityResourceClaimsTypes.Sub));
+
+            providerService = new Mock<IProviderService>();
+            providerController = new ProviderController(providerService.Object, localizer.Object, new Mock<ILogger<ProviderController>>().Object);
+
+
+
+            providerController.ControllerContext.HttpContext = new DefaultHttpContext { User = user };
+            providerDtos = ProviderDtoGenerator.Generate(10);
+            providerDto = ProviderDtoGenerator.Generate();
+        }
+
+        [Test]
+        public async Task GetProfile_WhenNoProviderWithSuchUserId_ReturnsNoContent()
+        {
+            // Arrange
+            ProviderDto nullProvider = null;
+            providerService.Setup(x => x.GetByUserId(It.IsAny<string>())).ReturnsAsync(nullProvider);
+
+            // Act
+            var result = await providerController.GetProfile().ConfigureAwait(false);
+
+            // Assert
+            Assert.IsInstanceOf<NoContentResult>(result);
+        }
+
+        [Test]
+        public async Task GetProfile_WhenProviderForUserIdExists_ReturnsOkObjectResult()
+        {
+            // Arrange
+            providerService.Setup(x => x.GetByUserId(It.IsAny<string>())).ReturnsAsync(providerDto);
+
+            // Act
+            var result = await providerController.GetProfile().ConfigureAwait(false);
+
+            // Assert
+            result.GetAssertedResponseOkAndValidValue<ProviderDto>();
+        }
+
+        [Test]
+        public async Task GetProviders_WhenCalled_ReturnsOkResultObject_WithColletionOfDTosValue()
+        {
+            // Arrange
+            providerService.Setup(x => x.GetAll()).ReturnsAsync(providerDtos);
+
+            // Act
+            var result = await providerController.Get().ConfigureAwait(false);
+
+            // Assert
+            result.GetAssertedResponseOkAndValidValue<IEnumerable<ProviderDto>>();
+        }
+
+
+        // write test to compare collection returned
+        [Test]
+        public async Task GetProviders_ReturnsExpectedCollectionOfDtos()
+        {
+            // Arrange
+            providerService.Setup(x => x.GetAll()).ReturnsAsync(providerDtos);
+
+
+            // Act
+            var result = (await providerController.Get().ConfigureAwait(false) as ObjectResult).Value as IEnumerable<ProviderDto>;
+
+            // Assert
+            Assert.That(providerDtos.Union(result).Count, Is.EqualTo(providerDtos.Count()));
```

Comment:
What does count exactly says here ? 
We should check all entries for equality.
Not sure if order is matter.

Reply:
Union method returns IEnumerable based on two other collections members, ignoring duplicates. Due to that logic, as we have same collection length in result Value as expected collection count is, my idea was that union of two totally similar collections produces third exactly same collection)

Reply:
ok, does that mean that 2 collections with the same number of entries are equals ?